src/wp-includes/class-wp-post-type.php

```diff
@@ -922,7 +923,7 @@ public function get_autosave_rest_controller() {
 			return null;
 		}
 
-		if ( 'attachment' === $this->name ) {
+		if ( in_array( $this->name, array( 'attachment', 'wp_global_styles' ), true ) ) {
```

Comment:
For consistency, it would be better to solve this similarly as for all the other post types introduced in 6.5, when registering the post type:

```
'autosave_rest_controller_class' => 'stdClass',
```

That said, in WordPress 6.6 I hope to make it easier to remove autosave support for a post type, see https://core.trac.wordpress.org/ticket/41172

Then all that's needed is `remove_post_type_support( 'wp_global_styles', autosave' )`, and no change in this method here.

Reply:
That handle this in https://core.trac.wordpress.org/ticket/41172 / 6.6. This fix work well.

Reply:
What I meant is:

- If this is merged in 6.5, let's use the `'autosave_rest_controller_class' => 'stdClass'` approach instead of changing this method, for consistency reasons.
- If this is merged in 6.6, let's use the `remove_post_type_support( 'wp_global_styles', autosave' )` approach for consistency reasons. That way, this method will not have post type specific logic in it anymore.

Reply:
I've added `'autosave_rest_controller_class' => 'stdClass'` in https://github.com/WordPress/wordpress-develop/pull/6573 for now because I couldn't decide where to call `remove_post_type_support( 'wp_global_styles', autosave' )`. 😄 Where would be a logical place to do this? In `wp_includes/post.php` I'm imagining.

Reply:
> Looking at the docs for remove_post_type_support autosave does not seem to be supported.

That's what https://core.trac.wordpress.org/ticket/41172 is about. It adds this.

Check out https://github.com/WordPress/wordpress-develop/pull/6035

@ramonjd That would be in `create_initial_post_types`, as per the above PR.

But again, this requires the above PR to be merged!

----

Back to my original comment:

What I don't like about `if ( in_array( $this->name, array( 'attachment', 'wp_global_styles' ), true ) )` is that we'd then have 2 different ways for disabling autosaves in core, which is not good.

For consistency, we should use the same approach as we already use elsewhere: `'autosave_rest_controller_class' => 'stdClass'`

OR, we just merge https://core.trac.wordpress.org/ticket/41172 because it's super close and makes all our lives easier.

Reply:
@swissspidy So you recommend just reverting back to `if ( 'attachment' === $this->name ) {` and `'autosave_rest_controller_class' => 'stdClass'`?

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/6105/commits/7c526bd178cde4fa54a7a22dcefeecf0599c971c

Reply:
https://core.trac.wordpress.org/ticket/41172 is now committed, so a better fix can be put in place.

Reply:
I've done this over in https://github.com/WordPress/wordpress-develop/pull/6573/files#diff-ef9c5475561d459917dac57d21e08a6dfeb36f8b1c44b978d3696aec8b97c405R509